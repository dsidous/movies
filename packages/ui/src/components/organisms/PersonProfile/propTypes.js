import { object } from 'prop-types';

export const propTypes = {
  person: object.isRequired,
};

export const defaultProps = {};
