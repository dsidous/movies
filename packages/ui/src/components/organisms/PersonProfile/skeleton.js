import React from 'react';

import Skeleton from '@material-ui/lab/Skeleton';
import Typography from '@material-ui/core/Typography';

import useStyles from './styles';

export default () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <div className={classes.infoRoot}>
        <div className={classes.poster}>
          <Skeleton variant="rect" height={300} />
        </div>
        <div className={classes.info}>
          <Typography variant="h2">
            <Skeleton variant="text" />
          </Typography>
          <Skeleton
            variant="text"
            width="20%"
            style={{ marginBottom: '24px' }}
          />
          <Typography variant="h6">
            <Skeleton
              variant="text"
              width="20%"
              style={{ marginBottom: '24px' }}
            />
          </Typography>
          <Skeleton variant="text" />
          <Skeleton variant="text" />
          <Skeleton variant="text" />
          <Skeleton variant="text" />
          <Skeleton variant="text" />
          <Skeleton variant="text" width="80%" />
        </div>
      </div>
    </div>
  );
};
