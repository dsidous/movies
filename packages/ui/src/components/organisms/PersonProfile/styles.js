import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      margin: '20px 10px',

      [theme.breakpoints.up('sm')]: {
        margin: '40px auto',
        maxWidth: 980,
        padding: '0 20px',
      },
    },

    infoRoot: {
      display: 'grid',
      gridTemplateColumns: '1fr 2fr',
      gridTemplateAreas: '"poster poster" "info info" "bio bio"',
      gridGap: '10px',

      [theme.breakpoints.up('sm')]: {
        gridColumnGap: 40,
        gridTemplateColumns: '1fr 2fr',
        gridTemplateRows: '100px auto',
        gridTemplateAreas: '"poster info" "poster bio"',
        margin: '0 0 40px',
      },
    },

    poster: {
      gridArea: 'poster',

      '& img': {
        borderRadius: '4px',
        boxShadow: '0 0 20px -5px #000',
        height: 'auto',
        maxWidth: '100%',
      },

      '& a': {
        display: 'block',
        color: '#000',
        position: 'relative',

        '&::after': {
          background: 'rgba(0,0,0,.8)',
          content: '" "',
          height: '100%',
          left: 0,
          opacity: 0,
          position: 'absolute',
          top: 0,
          transition: 'all linear .3s',
          visibility: 'hidden',
          width: '100%',
        },

        '&:hover::after': {
          opacity: 1,
          visibility: 'visible',
        },

        '&::before': {
          content: '"Show images"',
          color: '#fff',
          fontSize: '2em',
          left: '50%',
          position: 'absolute',
          textAlign: 'center',
          top: '50%',
          transform: 'translate(-50%,-50%)',
          visibility: 'hidden',
          zIndex: 2,
        },

        '&:hover::before': {
          visibility: 'visible',
        },
      },
    },

    info: {
      gridArea: 'info',
    },

    name: {
      fontWeight: '$fw-n',
      marginBottom: 0,
      marginTop: 0,

      [theme.breakpoints.up('sm')]: {
        fontSize: '4em',
        position: 'relative',

        '&::after': {
          color: 'rgba(170,170,170,.2)',
          content: 'attr(data-name)',
          fontSize: '1.3em',
          fontWeight: 'bold',
          left: '-20px',
          position: 'absolute',
          top: '-20px',
        },
      },
    },

    bioRoot: {
      gridArea: 'bio',
    },

    bio: {
      whiteSpace: 'pre-line',

      '&.long::after': {
        content: '" ..."',
      },

      '& label': {
        background: '#000',
        color: '#bbb',
        cursor: 'pointer',
        float: 'right',
        fontSize: 10,
        margin: '40px 10px 0 0',
        padding: '5px 10px',

        '&::before': {
          content: '"More"',
        },
      },
    },

    hidden: {
      opacity: 0,
      maxHeight: 0,
      fontSize: 0,
      transition: '0.25s ease',
      whiteSpace: 'normal',
    },

    more: {
      display: 'none',

      '&:checked ~ $bio.long::after': {
        display: 'none',
      },

      '&:checked ~ $bio $hidden': {
        opacity: 1,
        fontSize: 'inherit',
        maxHeight: '999em',
        whiteSpace: 'pre-line',
      },

      '&:checked ~ $bio label::before': {
        content: '"Less"',
      },
    },

    showsRoot: {
      borderTop: '$separator',
      margin: '40px 0 0',

      '& h4': {
        padding: '20px 0',

        [theme.breakpoints.up('sm')]: {
          '&:last-of-type': {
            margin: '0 auto',
            width: '80%',
          },
        },
      },
    },

    knowns: {
      display: 'grid',
      gridGap: 10,
      gridTemplateColumns: 'repeat(2, 1fr)',

      [theme.breakpoints.up('sm')]: {
        gridGap: 40,
        gridTemplateColumns: 'repeat(4, 1fr)',
      },
    },

    shows: {
      display: 'grid',
      gridGap: 0,
      gridTemplateColumns: '1fr',

      [theme.breakpoints.up('sm')]: {
        margin: '0 auto',
        width: '80%',
      },
    },
  }),
  { name: 'person' },
);
