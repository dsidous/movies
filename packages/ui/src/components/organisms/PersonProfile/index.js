/* eslint-disable camelcase */
import React from 'react';
import { FormattedDate } from 'react-intl';
import Markdown from 'markdown-to-jsx';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import SEO from '../../atoms/SEO';
import MediaImage from '../../atoms/MediaImage';
import PersonShows from '../../molecules/PersonShows';
import PersonKnownFor from '../../molecules/PersonKnownFor';

const PersonProfile = ({
  person: {
    biography,
    birthday,
    deathday,
    name,
    place_of_birth,
    profile_path,
    combined_credits,
    id,
  },
}) => {
  let firstName;
  let lastName;

  if (name) {
    [firstName, ...lastName] = name.split(' ');
    lastName = lastName.join(' ');
  }

  return (
    <div>
      <SEO title={name} />
      <div className="person-wrapper">
        <div className="person-info-wrapper">
          <div className="person-poster">
            <Link to={`/person/${id}/images`}>
              <MediaImage
                className="img-responsive center-block"
                mediaType="profile"
                size={2}
                filePath={profile_path}
                name="poster"
              />
            </Link>
          </div>
          <div className="person-info">
            <h3 className="person-name" data-name={lastName}>
              {firstName}
            </h3>
            <div>
              {`${lastName}, ${place_of_birth}`}
              {birthday && (
                <span>
                  ,&nbsp;
                  <FormattedDate
                    value={birthday}
                    year="numeric"
                    month="long"
                    day="2-digit"
                  />
                </span>
              )}
              {deathday && (
                <span>
                  &nbsp;-&nbsp;
                  <FormattedDate
                    value={deathday}
                    year="numeric"
                    month="long"
                    day="2-digit"
                  />
                </span>
              )}
            </div>
          </div>
          {biography && (
            <div className="person-bio-wrapper">
              <h4>Biography</h4>
              <input
                type="checkbox"
                className="person-bio__more"
                id="person-bio__more"
              />
              <div
                className={[
                  'person-bio',
                  biography.length > 500 ? 'long' : '',
                ].join(' ')}
              >
                <Markdown options={{ forceInline: true }}>
                  {biography.slice(0, biography.indexOf(' ', 500))}
                </Markdown>
                <span className="person-bio__hidden">
                  <Markdown options={{ forceInline: true }}>
                    {biography.slice(biography.indexOf(' ', 500))}
                  </Markdown>
                </span>
                {biography.length > 500 && (
                  // eslint-disable-next-line jsx-a11y/label-has-associated-control
                  <label
                    htmlFor="person-bio__more"
                    className="person-bio__more-label"
                  />
                )}
              </div>
            </div>
          )}
        </div>
        <div className="person-movies-wrapper">
          <h4>Knonw For</h4>
          <div className="person-movies-known-grid">
            <PersonKnownFor combinedCredits={combined_credits} />
          </div>
          <h4>Acting</h4>
          <div className="person-movies-grid">
            <PersonShows shows={combined_credits} />
          </div>
        </div>
      </div>
    </div>
  );
};

PersonProfile.propTypes = propTypes;

export default PersonProfile;
