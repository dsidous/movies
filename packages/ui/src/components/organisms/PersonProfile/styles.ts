import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  (theme) => ({
    root: {},
    infoRoot: {
      display: 'grid',
      gridTemplateColumns: '1fr 2fr',
      gridTemplateAreas: '"poster poster" "info info" "bio bio"',
      gridGap: theme.spacing(2),

      [theme.breakpoints.up('sm')]: {
        gridColumnGap: theme.spacing(5),
        gridTemplateColumns: '1fr 2fr',
        gridTemplateRows: '100px auto',
        gridTemplateAreas: '"poster info" "poster bio"',
        margin: theme.spacing(0, 0, 5),
      },
    },

    poster: {
      gridArea: 'poster',

      '& img': {
        borderRadius: theme.shape.borderRadius,
        boxShadow: '0 0 20px -5px #000',
        height: 'auto',
        maxWidth: '100%',
      },

      '& a': {
        display: 'block',
        color: '#000',
        position: 'relative',

        '&::after': {
          background: 'rgba(0,0,0,.8)',
          content: '" "',
          height: '100%',
          left: 0,
          opacity: 0,
          position: 'absolute',
          top: 0,
          transition: 'all linear .3s',
          visibility: 'hidden',
          width: '100%',
        },

        '&:hover::after': {
          opacity: 1,
          visibility: 'visible',
        },

        '&::before': {
          content: '"Show images"',
          color: '#fff',
          fontSize: '2em',
          left: '50%',
          position: 'absolute',
          textAlign: 'center',
          top: '50%',
          transform: 'translate(-50%,-50%)',
          visibility: 'hidden',
          zIndex: 2,
        },

        '&:hover::before': {
          visibility: 'visible',
        },
      },
    },

    info: {
      gridArea: 'info',
    },

    name: {
      fontWeight: theme.fontWeightLight,
      marginBottom: 0,
      marginTop: 0,

      [theme.breakpoints.up('sm')]: {
        fontSize: '4em',
        position: 'relative',

        '&::after': {
          color: 'rgba(170,170,170,.2)',
          content: 'attr(data-name)',
          fontSize: '1.3em',
          fontWeight: theme.fontWeightBold,
          left: theme.spacing(-3),
          position: 'absolute',
          top: theme.spacing(-3),
        },
      },
    },

    bioRoot: {
      gridArea: 'bio',
      marginTop: theme.spacing(2),
    },

    bio: {
      display: 'flex',
      flexDirection: 'column',
      whiteSpace: 'pre-line',
    },

    knowns: {
      display: 'grid',
      gridGap: theme.spacing(2),
      gridTemplateColumns: 'repeat(2, 1fr)',

      [theme.breakpoints.up('sm')]: {
        gridGap: theme.spacing(5),
        gridTemplateColumns: 'repeat(5, 1fr)',
      },
    },

    shows: {
      display: 'grid',
      gridGap: 0,
      gridTemplateColumns: '1fr',

      [theme.breakpoints.up('sm')]: {
        margin: '0 auto',
      },
    },
  }),
  { name: 'person' }
);
