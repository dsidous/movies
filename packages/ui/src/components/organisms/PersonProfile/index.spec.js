import React from 'react';
import { shallow } from 'enzyme';
import PersonProfile from '.';

const mockProps = {
  person: {
    biography: 'bio',
    birthday: '01.01.2000',
    deathday: '02.02.2010',
    name: 'name',
    place_of_birth: 'city',
    profile_path: 'path/to/image',
    combined_credits: {},
    id: 1,
  },
};

describe('Organisms/PersonProfile', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<PersonProfile {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
