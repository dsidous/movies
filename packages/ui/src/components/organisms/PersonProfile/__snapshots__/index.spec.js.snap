// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/PersonProfile should render as expected 1`] = `ShallowWrapper {}`;
